ci-operator/step-registry/osd-ccs/cluster/provision/gcp/osd-ccs-cluster-provision-gcp-commands.sh

```diff
@@ -101,56 +200,39 @@ ocm create cluster ${CLUSTER_NAME} \
 --subscription-type ${SUBSCRIPTION_TYPE} \
 ${MARKETPLACE_GCP_TERMS_SWITCH} \
 ${DISABLE_WORKLOAD_MONITORING_SWITCH} \
-${ETCD_ENCRYPTION_SWITCH}
-"
+${ETCD_ENCRYPTION_SWITCH} --secure-boot-for-shielded-vms"
+
+if [[ -v IS_XPN ]]; then
+  cmd="${cmd} --vpc-project-id ${VPC_PROJECT_ID} --vpc-name ${VPC_NAME} --control-plane-subnet ${CONTROL_PLANE_SUBNET} --compute-subnet ${COMPUTE_SUBNET}"
+fi
 
 # Create GCP cluster
-ocm create cluster ${CLUSTER_NAME} \
-                    --ccs \
-                    --provider=gcp \
-                    --region "${REGION}" \
-                    --service-account-file "${GCP_CREDENTIALS_FILE}" \
-                    --version "${OPENSHIFT_VERSION}" \
-                    --channel-group "${CHANNEL_GROUP}" \
-                    --compute-machine-type "${COMPUTE_MACHINE_TYPE}" \
-                    --subscription-type "${SUBSCRIPTION_TYPE}" \
-                    ${MARKETPLACE_GCP_TERMS_SWITCH} \
-                    ${DISABLE_WORKLOAD_MONITORING_SWITCH} \
-                    ${ETCD_ENCRYPTION_SWITCH} \
-                    > "${ARTIFACT_DIR}/cluster.txt"
+echo -e "$(date -u --rfc-3339=seconds) - INFO: Running Command '${cmd}'"
+eval "${cmd}" > "${ARTIFACT_DIR}/cluster.txt"
 
 # Store the cluster ID for the post steps and the cluster deprovision
 mkdir -p "${SHARED_DIR}"
 CLUSTER_ID=$(cat "${ARTIFACT_DIR}/cluster.txt" | grep '^ID:' | tr -d '[:space:]' | cut -d ':' -f 2)
-echo "Cluster ${CLUSTER_NAME} is being created with cluster-id: ${CLUSTER_ID}"
+echo "$(date -u --rfc-3339=seconds) - INFO: Cluster ${CLUSTER_NAME} is being created with cluster-id: ${CLUSTER_ID}"
 echo -n "${CLUSTER_ID}" > "${SHARED_DIR}/cluster-id"
 echo "${CLUSTER_NAME}" > "${SHARED_DIR}/cluster-name"
 
-echo "Waiting for cluster ready..."
-start_time=$(date +"%s")
-while true; do
-  sleep 60
-  CLUSTER_STATE=$(ocm get cluster "${CLUSTER_ID}" | jq -r '.status.state')
-  echo "Cluster state: ${CLUSTER_STATE}"
-  if [[ "${CLUSTER_STATE}" == "ready" ]]; then
-    echo "Cluster ${CLUSTER_ID} is reported as ready"
-    break
-  fi
-  if (( $(date +"%s") - $start_time >= $CLUSTER_TIMEOUT )); then
-    echo "error: Timed out while waiting for cluster to be ready"
-    exit 1
-  fi
-  if [[ "${CLUSTER_STATE}" != "installing" && "${CLUSTER_STATE}" != "pending" ]]; then
-    ocm get "/api/clusters_mgmt/v1/clusters/${CLUSTER_ID}/logs/install" > "${ARTIFACT_DIR}/.cluster_install.log" || echo "error: Unable to pull installation log."
-    echo "error: Cluster reported invalid state: ${CLUSTER_STATE}"
-    exit 1
-  fi
-done
+if [[ -v IS_XPN ]]; then
+  wait_for_cluster_state "${CLUSTER_ID}" "waiting"
+
+  echo "$(date -u --rfc-3339=seconds) - INFO: Granting the required permissions in the host project..."
+  ephemeral_sa_email=$(ocm describe cluster "${CLUSTER_ID}" | grep -Po "osd-managed-admin-[^\s\t]+")
+  add_iam_policy_binding "${ephemeral_sa_email}" "${VPC_PROJECT_ID}"
+
+  wait_for_cluster_state "${CLUSTER_ID}" "installing"
```

Comment:
It make the logic complex. I think we just need to add `if IS_XPN && CLUSTER_STATE=waiting; do add_iam_policy_binding` in the `waiting cluster ready` loop.